nntrainer/tensor/half_tensor.cpp

```diff
@@ -0,0 +1,2292 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2023 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * @file	 half_tensor.cpp
+ * @date	 09 November 2023
+ * @brief	 This is HalfTensor class for calculation
+ * @see		 https://github.com/nnstreamer/nntrainer
+ * @author	Donghyeon Jeong <dhyeon.jeong@samsung.com>
+ * @bug		 No known bugs except for NYI items
+ */
+
+#include <algorithm>
+#include <assert.h>
+#include <cmath>
+#include <cstring>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <iterator>
+#include <numeric>
+#include <regex>
+#include <sstream>
+#include <stdexcept>
+#include <stdio.h>
+
+#include <half_tensor.h>
+#include <util_func.h>
+
+#define transposeloop(cl, ci, cj, ck, sl, si, sj, sk)                 \
```

Comment:
This define can be shared through out the tensor. so.. place it in Tensor Class?

Reply:
You're right. In fact, I think the `transposeloop` and `transpose` functions should be in the future operation class (which includes multiply, sum, etc). For now, I left the operation functions intentionally for simplicity in verifying the operation class though it will be removed soon.

Reply:
Agreed. Thanks.